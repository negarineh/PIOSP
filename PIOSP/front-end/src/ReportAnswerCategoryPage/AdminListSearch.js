import React from "react";
import PropTypes from 'prop-types';
import AdminSearch from "./AdminSearch";
import AdminSearchCount from "./AdminSearchCount";
import { Search, Button, Segment, Pagination } from 'semantic-ui-react';

import {CSVLink} from 'react-csv';

const MATCHING_ITEM_LIMIT = 125;

// list of admins that we can have query between list create csv file
class AdminListSearch extends React.Component {
    constructor(props) { 
        super(props);   
        this.state={
            answers: [],
            showRemoveIcon: false,
            searchValue: "",
            activePage: 1,
            boundaryRange: 1,
            siblingRange: 1,
            showEllipsis: true,
            showFirstAndLastNav: true,
            showPreviousAndNextNav: true,
            totalPages: 10,
            count: 1,
            report:[],
        };
        this.handleSearchCancel = this.handleSearchCancel.bind(this);
        this.handleSearchChange = this.handleSearchChange.bind(this);
        this.handlePaginationChange = this.handlePaginationChange.bind(this);
      }

  handleSearchChange (e )  {

    e.preventDefault();
    const value = e.target.value;

    this.setState({
      searchValue: value,
      isLoading: true,
    });

    if (value === "") {
      this.setState({
        answers: [],
        showRemoveIcon: false,
      });
      } else {
        this.setState({
          showRemoveIcon: true
        });

      AdminSearch.search(value, this.state.activePage, 5, answers => {
        this.setState({ 
          answers: answers.slice(0, MATCHING_ITEM_LIMIT)
        });
      });

      AdminSearchCount.search(value, count => {
        this.setState({
          totalPages: count/5
        });
      });

      fetch(`/reportAdminActivity`,{
        method: 'POST',
        body: JSON.stringify({
          email: value
        }),
        headers: {"Content-Type": "application/json"}
      })
      .then(data => data.json())
      .then((data) => {this.setState({report : data});})
      .catch(err => console.error(err));
    }

    setTimeout(() => {

      this.setState({
        isLoading: false,
      });
    }, 500);
  }

  handleSearchCancel () {
    this.setState({
      answers: [],
      showRemoveIcon: false,
      searchValue: ""
    });
  }

  handlePaginationChange (e, { activePage }){
   
    e.preventDefault();
    this.setState({ activePage: activePage });
    
    AdminSearch.search(this.state.searchValue, activePage, 5, answers => {
      this.setState({
        answers: answers.slice(0, MATCHING_ITEM_LIMIT)
      });
    });
    }

  render() {
    const { answers, isLoading } = this.state;

    const answersRows = answers.map((answer, idx) => (
      <tr key={idx} >
        <td> <input 
                type="checkbox"
                name="checkbox" 
                className="form-control-12"
                ref={(a) => this.checkbox = a}
                required
                onClick={() => this.props.onAnswerClick(answer)}
                 /></td>
        <td className="left aligned" >{answer.Id}</td>
        <td className="left aligned" >{answer.email}</td>
        <td className="left aligned" >{answer.activityType}</td>
        <td className="left aligned" >{answer.timeOfActivity}</td>
    
      </tr>
    ));

    const {
      activePage,
      boundaryRange,
      siblingRange,
      showEllipsis,
      showFirstAndLastNav,
      showPreviousAndNextNav,
      totalPages,
    } = this.state
    return (
      <div id="food-search">
      <Segment>
        <table className="ui selectable structured  table">
          <thead>
            <tr>
              <th colSpan="5">
                <label style={{margin:'20px'}}>Enter Admin Email:</label>
                  <Button style={{float:'right', margin:'10px'}}><CSVLink data={answers} >Download this report</CSVLink></Button>
                    <Search
                        loading={isLoading}
                        onSearchChange={this.handleSearchChange}
                        value={this.state.searchValue}
                        placeholder="Search Admin Name..."
                        showNoResults={false}
                        icon={null}
                        style={{display:'inline'}}
                    />
              </th>
            </tr>
            <tr>
              <Pagination
                activePage={activePage}
                boundaryRange={boundaryRange}
                onPageChange={this.handlePaginationChange}
                size='mini'
                siblingRange={siblingRange}
                totalPages={totalPages}
                // All items are powered by shorthands, if you want to hide one of them, just pass `null` as value
                ellipsisItem={showEllipsis ? undefined : null}
                firstItem={showFirstAndLastNav ? undefined : null}
                lastItem={showFirstAndLastNav ? undefined : null}
                prevItem={showPreviousAndNextNav ? undefined : null}
                nextItem={showPreviousAndNextNav ? undefined : null}
              />
            </tr>
            <tr>
              <th>Select</th>
              <th>Id</th>
              <th>Email</th>
              <th>Activity Type</th>
              <th>Time of Activity</th>
            </tr>
          </thead>
          <tbody> 
          {answersRows}
          </tbody>
        </table>
        </Segment>
      </div>
    );
  }
}

export  {  AdminListSearch };

AdminListSearch.defaultProps = {
  onAnswerClick: () => {},
  onSearchChange: () => {},
  answers: {},
  isLoading: false,
  showRemoveIcon: false
}

AdminListSearch.propTypes = {
  onAnswerClick: PropTypes.func.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  answers: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  showRemoveIcon: PropTypes.bool.isRequired
}
