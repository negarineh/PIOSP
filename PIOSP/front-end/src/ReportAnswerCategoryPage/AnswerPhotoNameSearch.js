import React from "react";
import PropTypes from 'prop-types';
import NameSearch from "./NameSearch";
import {Button, Search, Segment, Pagination } from 'semantic-ui-react';

import {CSVLink} from 'react-csv';


const MATCHING_ITEM_LIMIT = 125;
var globalResponse = 0;

//search base on photos name and export as a csv file
class AnswerPhotoNameSearch extends React.Component {
    constructor(props) {
        super(props);   
        this.state={
            answers: [],
            showRemoveIcon: false,
            searchValue: "",
            activePage: 1,
            boundaryRange: 1,
            siblingRange: 1,
            showEllipsis: true,
            showFirstAndLastNav: true,
            showPreviousAndNextNav: true,
            totalPages: 10,
            count: 1,
        };
        this.handleSearchCancel = this.handleSearchCancel.bind(this);
        this.handleSearchChange = this.handleSearchChange.bind(this);
        this.handlePaginationChange = this.handlePaginationChange.bind(this);
      }

  handleSearchChange (e )  {

    e.preventDefault();
    const value = e.target.value;
      
    this.setState({
      searchValue: value,
      isLoading: true,
    });

    if (value === "") {
      this.setState({
        answers: [],
        showRemoveIcon: false,
      });
    } else {
      this.setState({
        showRemoveIcon: true
      });

      NameSearch.search(value, this.state.activePage, 5, answers => {
        this.setState({
          answers: answers.slice(0, MATCHING_ITEM_LIMIT)
        });
      });

      fetch(`reportAnswersPhotoIdCount`,{
        method: 'POST',
        body:JSON.stringify({
          photoName: value
        }),
        headers: {"Content-Type": "application/json"}
      })
      .then(data => data.json())
      .then((data) => { globalResponse = data;
      console.log(globalResponse); // eslint-disable-line no-console
    }).catch(err => console.log(err));

  this.setState({totalPages:globalResponse/5});
    }
    setTimeout(() => {

      this.setState({
        isLoading: false,
      })
    }, 500)
  }

  handleSearchCancel () {
    this.setState({
      answers: [],
      showRemoveIcon: false,
      searchValue: ""
    });
  }

  handlePaginationChange (e, { activePage }){
  
    e.preventDefault();
    this.setState({ activePage: activePage });

    NameSearch.search(this.state.searchValue, activePage, 5, answers => {
      this.setState({
        answers: answers.slice(0, MATCHING_ITEM_LIMIT)
      });
    });
    }

  render() {
    const { answers, isLoading } = this.state;
    // const removeIconStyle = showRemoveIcon ? {} : { visibility: "hidden" };

    const answersRows = answers.map((answer, idx) => (
      <tr key={idx} >
           <td><input 
                type="checkbox"
                name="checkbox" 
                className="form-control-12"
                ref={(a) => this.checkbox = a}
                required
                onClick={() => this.props.onAnswerClick(answer)}
                 /></td> 
        <td className="left aligned" >{answer.photo}</td>
        <td className="left aligned" >{answer.answer}</td>
        <td className="left aligned" >{answer.description}</td>
      </tr>
    ));

    const {
      activePage,
      boundaryRange,
      siblingRange,
      showEllipsis,
      showFirstAndLastNav,
      showPreviousAndNextNav,
      totalPages,
    } = this.state

    return (
      <div id="search">
        <Segment>
          <table className="ui selectable structured table">
              <thead>
                <tr>
                  <th colSpan="5">
                    <label style={{margin:'20px'}}>Enter photo Name:</label>
                      <Button style={{float:'right', margin:'10px'}}><CSVLink data={answers} >Download this report</CSVLink></Button>
                        <Search
                            loading={isLoading}
                            // onResultSelect={this.handleResultSelect}
                            onSearchChange={this.handleSearchChange}
                            // results={results}
                            value={this.state.searchValue.charAt(0).toUpperCase() + this.state.searchValue.slice(1)}
                            placeholder="Search answers..."
                            // {...this.props}
                            showNoResults={false}
                            icon={null}
                            style={{display:'inline'}}
                        />
                        {/* <i className="search icon" />
                        <i
                          className="remove icon"
                          onClick={this.handleSearchCancel}
                          style={removeIconStyle}
                        /> */}
                    </th>
                  </tr>
                  <tr style={{textAlign:'center'}}>
                    <Pagination
                      activePage={activePage}
                      boundaryRange={boundaryRange}
                      onPageChange={this.handlePaginationChange}
                      size='mini'
                      siblingRange={siblingRange}
                      totalPages={totalPages}
                      // Heads up! All items are powered by shorthands, if you want to hide one of them, just pass `null` as value
                      ellipsisItem={showEllipsis ? undefined : null}
                      firstItem={showFirstAndLastNav ? undefined : null}
                      lastItem={showFirstAndLastNav ? undefined : null}
                      prevItem={showPreviousAndNextNav ? undefined : null}
                      nextItem={showPreviousAndNextNav ? undefined : null}
                    />
                  </tr>
                  <tr >
                    <th>Select</th>
                    <th>photo</th>
                    <th>Answer</th>
                    <th>Description</th>
                    {/* <th>_id</th> */}
                  </tr>
              </thead>
            <tbody> 
            {answersRows}
            </tbody>
          </table>
        </Segment>
      </div>
    );
  }
}

// export default ReportAnswerCategoryPage;
export  {  AnswerPhotoNameSearch };

AnswerPhotoNameSearch.defaultProps = {
  onAnswerClick: () => {},
  handleSearchChange: () => {},
  handleSearchCancel: () => {},
  answers: {},
  isLoading: false,
  showRemoveIcon: false
}

AnswerPhotoNameSearch.propTypes = {
  onAnswerClick: PropTypes.func.isRequired,
  handleSearchChange: PropTypes.func.isRequired,
  handleSearchCancel: PropTypes.func.isRequired,
  answers: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  showRemoveIcon: PropTypes.bool.isRequired
}