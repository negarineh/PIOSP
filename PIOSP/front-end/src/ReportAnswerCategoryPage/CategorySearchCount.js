
/* eslint-disable no-undef */
function search(query, cb) {
    return fetch(`reportAnswersCategoryCount`, { 
      headers: {"Content-Type": "application/json"},
      method: "POST",
      body:JSON.stringify({
              answer: query,
            }),
    })
      .then(checkStatus)
      .then(parseJSON)
      .then(cb)
      .catch(err => console.log(err));
  }
  
  function checkStatus(response) {
    if (response.status >= 200 && response.status < 300) {
      return response;
    }
    const error = new Error(`HTTP Error ${response.statusText}`);
    error.status = response.statusText;
    error.response = response;
    console.log(error); // eslint-disable-line no-console
    throw error;
  }
  
  function parseJSON(response) {
    return response.json();
  }
  
  const CategorySearchCount = { search };
  export default CategorySearchCount;