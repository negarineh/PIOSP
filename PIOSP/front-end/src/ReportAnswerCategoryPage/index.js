export * from './ReportAnswerCategoryPage';
export * from './CategorySearch';
export * from './SelectedAnswers';
export * from './AnswerCategorySearch';
export * from './NameSearch';
export * from './AdminSearch';
export * from './AdminListSearch';
export * from './SelectedAdmin';
export * from './ReportAnswerPhotoPage';
export * from './ReportAdminPage';