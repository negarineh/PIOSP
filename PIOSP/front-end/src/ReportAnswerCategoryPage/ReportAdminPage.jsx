import React from "react";
import PropTypes from 'prop-types';
import SelectedAdmin from "./SelectedAdmin";
import {AdminListSearch} from './AdminListSearch';
import { Segment } from "semantic-ui-react";


// includes selected admin class and admin list search component for create report(.csv)
class ReportAdminPage extends React.Component {
  state = {
    selectedAnswers: []
  };

  removeAnswerItem = itemIndex => {
    const filteredAnswers = this.state.selectedAnswers.filter(
      (item, idx) => itemIndex !== idx
    );
    this.setState({ selectedAnswers: filteredAnswers });
  };

  addAnswer = Answer => {
    const newAnswers = this.state.selectedAnswers.concat(Answer);
    this.setState({ selectedAnswers: newAnswers });
  };

  render() {
    const { selectedAnswers } = this.state;

    return ( 
      <div className="App">
      <Segment style={{height: '638.5px'}}>
          <SelectedAdmin
            answers={selectedAnswers}
            onAnswerClick={this.removeAnswerItem}
          />
          <AdminListSearch onAnswerClick={this.addAnswer} />
        </Segment>
      </div>
    );
  }
}

// export default ReportAnswerCategoryPage;
export {  ReportAdminPage };

ReportAdminPage.defaultProps = {
  removeAnswerItem: () => {},
  addAnswer: () => {}
}

ReportAdminPage.propTypes = {
  removeAnswerItem: PropTypes.func.isRequired,
  addAnswer: PropTypes.func.isRequired,
}
