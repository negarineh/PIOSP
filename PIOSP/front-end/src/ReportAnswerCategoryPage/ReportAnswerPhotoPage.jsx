import React from "react";
import PropTypes from 'prop-types';
import SelectedAnswers from "./SelectedAnswers";
import {AnswerPhotoNameSearch} from './AnswerPhotoNameSearch';
import { Segment } from "semantic-ui-react";

// includes selected answers class and answers photo name search component for create report(.csv)
class ReportAnswerPhotoPage extends React.Component {
  state = {
    selectedAnswers: []
  };

  removeAnswerItem = itemIndex => {
    const filteredAnswers = this.state.selectedAnswers.filter(
      (item, idx) => itemIndex !== idx
    );
    this.setState({ selectedAnswers: filteredAnswers });
  };

  addAnswer = Answer => {
    const newAnswers = this.state.selectedAnswers.concat(Answer);
    this.setState({ selectedAnswers: newAnswers });
  };

  render() {
    const { selectedAnswers } = this.state;

    return ( 
      <div className="App">
        <Segment style={{height: '638.5px'}}>
          <SelectedAnswers
            answers={selectedAnswers}
            onAnswerClick={this.removeAnswerItem}
          />
          <AnswerPhotoNameSearch onAnswerClick={this.addAnswer} />
        </Segment>
      </div>
    );
  }
}

// export default ReportAnswerCategoryPage;
export {  ReportAnswerPhotoPage };

ReportAnswerPhotoPage.defaultProps = {
  removeAnswerItem: () => {},
  addAnswer: () => {},
}

ReportAnswerPhotoPage.propTypes = {
  removeAnswerItem: PropTypes.func.isRequired,
  addAnswer: PropTypes.func.isRequired,
}
