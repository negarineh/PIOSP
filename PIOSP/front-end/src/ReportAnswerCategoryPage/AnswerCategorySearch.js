import React from "react";
import PropTypes from 'prop-types';
import CategorySearch from "./CategorySearch";
import CategorySearchCount from './CategorySearchCount';
import {Button, Search, Segment, Pagination } from 'semantic-ui-react';

import {CSVLink} from 'react-csv';
import '../components/stylesheets/Page3.css';

const MATCHING_ITEM_LIMIT = 125;

// search base on pollinator category between answers
// and export as a csv file
class AnswerCategorySearch extends React.Component {
    constructor(props) {
        super(props);   
        this.state={
            answers: [],
            showRemoveIcon: false,
            searchValue: "",
            activePage: 1,
            boundaryRange: 1,
            siblingRange: 1,
            showEllipsis: true,
            showFirstAndLastNav: true,
            showPreviousAndNextNav: true,
            totalPages: 10,
            count: 1,
            report:[],
        };
        this.handleSearchCancel = this.handleSearchCancel.bind(this);
        this.handleSearchChange = this.handleSearchChange.bind(this);
        this.handlePaginationChange = this.handlePaginationChange.bind(this);
        this.handleReportButton = this.handleReportButton.bind(this);
      }

  handleSearchChange (e )  {

    e.preventDefault();
    const value = e.target.value;

    this.setState({
      searchValue: value,
      isLoading: true,
    });

    if (value === "") {
      this.setState({
        answers: [],
        showRemoveIcon: false,
      });
    } else { 
      this.setState({
        showRemoveIcon: true
      });

      CategorySearch.search(value, this.state.activePage, 5, answers => {
        this.setState({
          answers: answers.slice(0, MATCHING_ITEM_LIMIT)
        });
      });
      CategorySearchCount.search(value, count => {
        this.setState({
          totalPages: count/5
        });
      });

      fetch(`/reportOnAnswersCategory`,{
        method: 'POST',
        body: JSON.stringify({
          answer: value
        }),
        headers: {"Content-Type": "application/json"}
      })
      .then(data => data.json())
      .then((data) => {this.setState({report : data});})
      .catch(err => console.error(err));
    }

    setTimeout(() => {

      this.setState({
        isLoading: false,
      })
    }, 500)
  }

  handleSearchCancel () {
    this.setState({
      answers: [],
      showRemoveIcon: false,
      searchValue: ""
    });
  }

handlePaginationChange (e, { activePage }){
   
e.preventDefault();
this.setState({ activePage: activePage });

CategorySearch.search(this.state.searchValue, activePage, 5, answers => {
  this.setState({
    answers: answers.slice(0, MATCHING_ITEM_LIMIT)
  });
});
}

handleReportButton(e){
  e.preventDefault();
  const value = e.target.value;
  fetch(`/reportOnAnswersCategory`,{
    method: 'POST',
    body: JSON.stringify({
      answer: value
    }),
    headers: {"Content-Type": "application/json"}
  })
  .then(data => data.json())
  .then((data) => {this.setState({report : data});})
  .catch(err => console.error(err));
}

  /* jshint ignore:start */
  render() {
    const { answers, isLoading } = this.state;

    const answersRows = answers.map((answer, idx) => (
      <tr key={idx}>
        <td> 
        <input 
                type="checkbox"
                name="checkbox" 
                className="form-control-12"
                ref={(a) => this.checkbox = a}
                required
                onClick={() => this.props.onAnswerClick(answer)} />
        </td>
        <td className="left aligned" >{answer.photo}</td>
        <td className="left aligned" >{answer.answer}</td>
        <td className="left aligned" >{answer.description}</td>
      </tr>
    ));
  /* jshint ignore:end */

    const {
      activePage,
      boundaryRange,
      siblingRange,
      showEllipsis,
      showFirstAndLastNav,
      showPreviousAndNextNav,
      totalPages,
    } = this.state

    return (
      <div id="search" >
        <Segment>
          <table className="ui selectable structured  table">
            <thead>
              <tr>
                <th colSpan="5">
                  <label style={{margin:'20px'}}>Enter photo category name:</label>
                      <Button style={{float:'right', margin:'10px'}}>
                        <CSVLink data={answers} >Download this report</CSVLink>
                        {/* <CSVDownload data={report} target="_blank" /> */}
                      </Button>
                          <Search
                              loading={isLoading}
                              // onResultSelect={this.handleResultSelect}
                              onSearchChange={this.handleSearchChange}
                              // results={results}
                              value={this.state.searchValue.charAt(0).toUpperCase() + this.state.searchValue.slice(1)}
                              placeholder="Search answers..."
                              // {...this.props}
                              showNoResults={false}
                              // size='large'
                              icon={null}
                              style={{display:'inline'}}
                          />
                          {/* <i className="search icon"/>
                          <i
                            className="remove icon"
                            onClick={this.handleSearchCancel}
                            style={removeIconStyle}
                          /> */}
                  </th>
              </tr>
              <tr>
              <Pagination
                activePage={activePage}
                boundaryRange={boundaryRange}
                onPageChange={this.handlePaginationChange}
                size='mini'
                siblingRange={siblingRange}
                totalPages={totalPages}
                // All items are powered by shorthands, if you want to hide one of them, just pass `null` as value
                ellipsisItem={showEllipsis ? undefined : null}
                firstItem={showFirstAndLastNav ? undefined : null}
                lastItem={showFirstAndLastNav ? undefined : null}
                prevItem={showPreviousAndNextNav ? undefined : null}
                nextItem={showPreviousAndNextNav ? undefined : null}
              />
              </tr>
              <tr >
                <th>Select</th>
                <th>photo</th>
                <th>Answer</th>
                <th>Description</th>
              </tr>
            </thead>
              <tbody style={{width:'10px'}}> 
                {answersRows}
              </tbody>
            </table>
          
          </Segment>
      </div>
    );
  }
}

// export default ReportAnswerCategoryPage;
export  {  AnswerCategorySearch };

AnswerCategorySearch.defaultProps = {
  onAnswerClick: () => {},
  handleSearchChange: () => {},
  handleSearchCancel: () => {},
  answers: {},
  isLoading: false,
  showRemoveIcon: false
}

AnswerCategorySearch.propTypes = {
  onAnswerClick: PropTypes.func.isRequired,
  handleSearchChange: PropTypes.func.isRequired,
  handleSearchCancel: PropTypes.func.isRequired,
  answers: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  showRemoveIcon: PropTypes.bool.isRequired
}
