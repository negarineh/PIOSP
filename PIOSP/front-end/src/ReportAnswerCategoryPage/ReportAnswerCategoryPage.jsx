import React from "react";
import PropTypes from 'prop-types';
import {AnswerCategorySearch} from "./AnswerCategorySearch";
import SelectedAnswers from "./SelectedAnswers";
import { Segment } from "semantic-ui-react";

// includes selected answers class and answers category search component for create report(.csv)
class ReportAnswerCategoryPage extends React.Component {
  state = {
    selectedAnswers: []
  };

  removeAnswerItem = itemIndex => {
    const filteredAnswers = this.state.selectedAnswers.filter(
      (item, idx) => itemIndex !== idx
    );
    this.setState({ selectedAnswers: filteredAnswers });
  };

  addAnswer = Answer => {
    const newAnswers = this.state.selectedAnswers.concat(Answer);
    this.setState({ selectedAnswers: newAnswers });
  };

  render() {
    const { selectedAnswers } = this.state;

    return ( 
      <div className="App">
        <Segment style={{height: '638.5px'}}>
          <SelectedAnswers
            answers={selectedAnswers}
            onAnswerClick={this.removeAnswerItem}
          />
          <AnswerCategorySearch onAnswerClick={this.addAnswer} />
        </Segment>
      </div>
    );
  }
}

export {  ReportAnswerCategoryPage };

ReportAnswerCategoryPage.defaultProps = {
  removeAnswerItem: () => {},
  addAnswer: () => {},
}

ReportAnswerCategoryPage.propTypes = {
  removeAnswerItem: PropTypes.func.isRequired,
  addAnswer: PropTypes.func.isRequired,
}
