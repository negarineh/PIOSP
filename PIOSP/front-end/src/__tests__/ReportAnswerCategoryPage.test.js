import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import {ReportAnswerCategoryPage} from "../ReportAnswerCategoryPage";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() }); 

// unit tests for the UserDeletePrompt component
describe('Report Answer Category Page component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {removeAnswerItem: ()=>{}, addAnswer: ()=>{}};
      const wrapper = shallow(<ReportAnswerCategoryPage {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
