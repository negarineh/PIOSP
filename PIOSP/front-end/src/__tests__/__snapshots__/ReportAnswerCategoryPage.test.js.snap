// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Answer Category Page component render() should render the component 1`] = `
<div
  className="App"
>
  <Segment
    style={
      Object {
        "height": "638.5px",
      }
    }
  >
    <SelectedAnswers
      answers={Array []}
      onAnswerClick={[Function]}
    />
    <AnswerCategorySearch
      answers={Object {}}
      handleSearchCancel={[Function]}
      handleSearchChange={[Function]}
      isLoading={false}
      onAnswerClick={[Function]}
      showRemoveIcon={false}
    />
  </Segment>
</div>
`;
